src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -726,6 +777,26 @@ def parent_construct_payload(self):
         if not managed_env_info:
             raise ValidationError("The environment '{}' does not exist. Specify a valid environment".format(self.get_argument_managed_env()))
 
+        consumption_gpu_wp = self.validate_consumption_gpu_profile()
+        if consumption_gpu_wp is not None:
+            logger.warning("Enabled preview feature: Consumption GPU workload profile.")
+            if self.get_argument_workload_profile_name() is not None:
+                raise ValidationError("Both --consumption-gpu-profile and --workload-profile-name (-w) are specified. Only one can be selected.")
+            existing_wp = safe_get(managed_env_info, "properties", "workloadProfiles", default=None)
+            consumption_gpu_wp_name = None
+            if existing_wp is not None:
+                for wp in existing_wp:
+                    if wp["workloadProfileType"].lower() == consumption_gpu_wp:
+                        consumption_gpu_wp_name = wp["name"]
+                        break
+            if consumption_gpu_wp_name is None:
+                env_client = self.get_environment_client
+                wp_payload, consumption_gpu_wp_name = self.update_consumption_gpu_wp(managed_env_info, consumption_gpu_wp)
+                env_client().update(cmd=self.cmd, resource_group_name=managed_env_rg, name=managed_env_name, managed_environment_envelope=wp_payload)
```

Comment:
If we update environment here, the `--no-wait` will not work for this case, because we have to wait for the update finish.

Reply:
We should warning customer the --no-wait not  take effect for this case:

--no-wait will not take effect when using --consumption-gpu-profile and need to create a workload profile.